tests/scene/test_packed_scene.h

```diff
@@ -280,4 +351,164 @@ TEST_CASE("[PackedScene] Recreate State") {
 	memdelete(scene);
 }
 
+TEST_CASE("[PackedScene] Preserve Array Types With Nil Default") {
+	GDREGISTER_CLASS(_TestIntArrayProperty);
+
+	// Create a scene to pack.
+	_TestIntArrayProperty *scene = memnew(_TestIntArrayProperty);
+	String property_name = "property";
+
+	// This test is pointless if default value can't be nil.
+	REQUIRE(scene->get(property_name).get_type() == Variant::NIL);
+
+	// Set property to a typed array.
+	Array array;
+	StringName class_name;
+	Ref<Script> script;
+	array.set_typed(Variant::INT, class_name, script);
+	array.push_back(0);
+	array.push_back(1);
+	array.push_back(2);
+	scene->set(property_name, array);
+
+	// Instantiate the scene.
+	PackedScene packed_scene;
```

Comment:
```suggestion
	Ref<PackedScene> packed_scene;
```
Should use ref semantics as it is reference counted, for all

Reply:
Also made a separate PR #112648 to do this for the rest of this file